sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -280,6 +296,23 @@ def tune(
                 + exec_script
             )
 
+        resources = client.V1ResourceRequirements()
+        if isinstance(resources_per_trial, dict):
+            requests = {
+                "cpu": "200m",
+                "memory": "256Mi",
+            }
```

Comment:
Also, I think we should avoid setting any default resources on behalf of the user.
If user sets only `gpu` in `resources_per_trial` dictionary, we should not set CPU and Memory, and resource allocation will be handle by Kubernetes.
WDYT @droctothorpe ?

Reply:
Agreed that would be preferable. I have to check what K8s does when you pass a `V1ResourceRequirement` with one key set but not the others. Will test and report back.

Reply:
Looks like K8s fills in the gaps! Great call, @andreyvelich. Simplifies the logic too. Changes pushed up. I think that covers all the comments. Standing by for next steps.

Here are the validation results:

`tune` call:

```python
katib_client.tune(
    name=name,
    objective=objective,
    parameters=parameters,
    objective_metric_name="result",
    max_trial_count=12,
    base_image=image,
    namespace=namespace,
    algorithm_name="random",
    resources_per_trial={
        "cpu": 2
    }
)
```

Experiment spec:
```yaml
              resources:
                limits:
                  cpu: 2
                requests:
                  cpu: 2
```

Pod spec:
```yaml
    resources:
      limits:
        cpu: "2"
        memory: 256Mi
      requests:
        cpu: "2"
        memory: 256Mi
```

Once more with feeling, but specifying memory instead of CPU:

`tune` call:

```python
katib_client.tune(
    name=name,
    objective=objective,
    parameters=parameters,
    objective_metric_name="result",
    max_trial_count=12,
    base_image=image,
    namespace=namespace,
    algorithm_name="random",
    resources_per_trial={
        "memory": "2Gi"
    }
)
```

Experiment spec:
```yaml
              resources:
                limits:
                  memory: 2Gi
                requests:
                  memory: 2Gi
```

Pod spec:
```yaml
    resources:
      limits:
        cpu: 200m
        memory: 2Gi
      requests:
        cpu: 200m
        memory: 2Gi
```